zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -1146,29 +1236,30 @@ where
                 SentTransactionOutput::from_parts(output_index, recipient, value, memo)
             });
 
-    let transparent_outputs =
-        transparent_output_meta
-            .into_iter()
-            .map(|(recipient, addr, value)| {
-                let script = addr.script();
-                let output_index = build_result
-                    .transaction()
-                    .transparent_bundle()
-                    .and_then(|b| {
-                        b.vout
-                            .iter()
-                            .enumerate()
-                            .find(|(_, tx_out)| tx_out.script_pubkey == script)
-                    })
-                    .map(|(index, _)| index)
-                    .expect(
-                        "An output should exist in the transaction for each transparent payment.",
-                    );
+    let txid: [u8; 32] = build_result.transaction().txid().into();
+    assert_eq!(
+        transparent_output_meta.len(),
+        build_result
+            .transaction()
+            .transparent_bundle()
+            .map_or(0, |b| b.vout.len()),
+    );
 
-                SentTransactionOutput::from_parts(output_index, recipient, value, None)
-            });
+    #[allow(unused_variables)]
+    let transparent_outputs = transparent_output_meta.into_iter().enumerate().map(
+        |(n, (recipient, ephemeral_address, value, step_output_index))| {
```

Comment:
This is misleading; `ephemeral_address` is only ephemeral for at most one entry in `transparent_output_meta`, because we also push into `transparent_output_meta` from inside the `add_transparent_output` closure. Rename this to either `to` or `address`.